Add the ability to add `extraMetrics` for HPA

anthosz opened the pull request:
In addition of cpu/memory basic metric, goal is to allow to scale the pods based on custom metrics (by example: the libreoffice/chromium queue metrics)

Tasks:

- [x] I've made changes
- [x] I've bumped chart's version in `Chart.yaml`
- [x] I've added changes to charts `CHANGELOG.md`
- [x] I've run [`helm-docs`](https://github.com/norwoodj/helm-docs)
- [x] I've run ['helm-tool schema | jq . > values.schema.json'](https://github.com/cert-manager/helm-tool)

MaikuMori commented on charts/gotenberg/values.yaml:
```diff
@@ -74,6 +74,7 @@ autoscaling:
   minReplicas: 1
   maxReplicas: 100
   behavior: {}
+  customMetrics: {}
```
```suggestion
  extraMetrics: {}
```

MaikuMori replied:
In fact this probably be an array of extra metrics.

```suggestion
  extraMetrics: []
```

MaikuMori commented on charts/gotenberg/templates/hpa.yaml:
```diff
@@ -31,4 +31,7 @@ spec:
     {{- end }}
+    {{- if .Values.autoscaling.customMetrics }}
+    {{- toYaml .Values.autoscaling.customMetrics | nindent 4 }}
+    {{- end }}
```
I think it should be array and handle it similarly to volumes.

MaikuMori commented on charts/gotenberg/Chart.yaml:
```diff
@@ -22,7 +22,7 @@ version: "1.9.1"
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "8.14.1"
+appVersion: "8.15.0"
```
Also update chart version.

MaikuMori commented on charts/gotenberg/templates/hpa.yaml:
```diff
@@ -31,4 +31,7 @@ spec:
           type: Utilization
           averageUtilization: {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}
     {{- end }}
+    {{- if .Values.autoscaling.customMetrics }}
```
```suggestion
    {{- if .Values.autoscaling.extraMetrics }}
```

MaikuMori commented on charts/gotenberg/templates/hpa.yaml:
```diff
@@ -31,4 +31,7 @@ spec:
           averageUtilization: {{ .Values.autoscaling.targetMemoryUtilizationPercentage }}
     {{- end }}
+    {{- if .Values.autoscaling.customMetrics }}
+    {{- toYaml .Values.autoscaling.customMetrics | nindent 4 }}
```
```suggestion
    {{- toYaml .Values.autoscaling.extraMetrics | nindent 4 }}
```

MaikuMori commented on charts/gotenberg/CHANGELOG.md:
```diff
@@ -1,5 +1,9 @@
 
+## 1.10.0
+
+- Adding the ability to add customMetrics in hpa
```
```suggestion
- Adding the ability to add customMetrics in hpa ([@anthosz](https://github.com/anthosz))
```

anthosz commented:
All done

MaikuMori commented:
Any chance you could enable this? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

I've cleaned up the pull request, but I can't push changes (I rebased and compressed it down to 2 commits).

Unless you have some custom action setup, in which case I can just push my cleanup to separate branch in this repo and merge that.

MaikuMori commented:
Okay I'm dumb-dumb. All is sorted out.

MaikuMori commented:
There was issue with pushing to OCI from release action. I manually pushed the same artifact that's available at releases link.

This should be fixed now for future. Also, what a silly error.

anthosz commented:
> Any chance you could enable this? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork
> 
> I've cleaned up the pull request, but I can't push changes (I rebased and compressed it down to 2 commits).
> 
> Unless you have some custom action setup, in which case I can just push my cleanup to a separate branch in this repo and merge that.
> 
> EDIT: I think it's already enabled. So I guess it doesn't like that I rebased, I don't get it.

I confirm that it was enabled

Anyway thank you!
